models/cam-se/model_mod.f90

```diff
@@ -3319,6 +3353,43 @@ end subroutine solve_quadratic
 
 !-----------------------------------------------------------------------
 
+subroutine verify_state_var_list
+
+! No return is needed.  A diagnostic statement is printed before a potential failure
+! of get_varid_from_varname.  Success means carry on.
+
+! There could be more moisture variables than the traditional 3.
+character(len=32), dimension(10) :: var_names
+integer:: i, varid
+
+data var_names /'PS', 'Q', 'CLDLIQ', 'CLDICE', 6*''/
```

Comment:
functionally this looks to be doing what is should, but it seems a little over complicated. 

Why do you have 10 var_names, but only 4 names, and 6 blank strings?

Why is PS in the var_names array? 
You could check for PS (always required), then loop though your sometimes required variables. 
That way 
```
i = 1
i = i+1
if (dry_mass_vertical_coordinate .and. precise_dry_mass_get_close) then
do while (var_names(i) .ne. '')
  i = i + 1
enddo
endif
```
becomes
```
check for PS
if (dry_mass_vertical_coordinate .and. precise_dry_mass_get_close) then
  do i = 1, 3
  enddo
endif
```